docs/aws/audit/dynamodbmonitoring/rules/kinesis_stream_encrypted_with_cmk_remediation.mdx

```diff
@@ -1,6 +1,216 @@
+           'StreamViewType': 'NEW_AND_OLD_IMAGES'
+       }
+   )
+
+   print(response)
+   ```
+
+By following these steps, you can ensure that your Kinesis Streams and DynamoDB Streams are not using a Customer Master Key (CMK) for encryption, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Kinesis console. You can do this by typing "Kinesis" into the search bar and selecting "Kinesis" from the dropdown menu.
+3. In the Kinesis console, select "Data Streams" from the navigation pane on the left side of the screen.
+4. Select the name of the stream you want to check. This will bring up the details of the stream.
+5. In the "Details" section, look for the "Encryption" field. If the stream is encrypted with a CMK, it will say "KMS". If it is not, it will say "None".
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access Kinesis Stream and DynamoDB.
+
+2. Once AWS CLI is installed and configured, you can use the following command to list all the Kinesis streams:
+
+   ```
+   aws kinesis list-streams
+   ```
+
+   This command will return a list of all the Kinesis streams in your AWS account.
+
+3. To check if a specific Kinesis stream is encrypted with a CMK, you can use the following command:
+
+   ```
+   aws kinesis describe-stream --stream-name your-stream-name
+   ```
+
+   Replace 'your-stream-name' with the name of your Kinesis stream. This command will return a JSON object that contains information about the stream.
+
+4. In the returned JSON object, look for the 'StreamDescription' field. Inside this field, there should be a 'StreamEncryptionType' field. If the value of this field is 'KMS', then the stream is encrypted with a CMK. If the value is 'NONE', then the stream is not encrypted. If the 'StreamEncryptionType' field is not present, then the stream is not encrypted.
+
+   Here is an example of what the 'StreamDescription' field might look like for a stream that is encrypted with a CMK:
+
+   ```
+   "StreamDescription": {
+       ...
+       "StreamEncryptionType": "KMS",
+       ...
+   }
+   ```
+
+   If the 'StreamEncryptionType' field is not present or if its value is 'NONE', then the stream is not encrypted.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>